src/core/gui/toolbarMenubar/ToolPageLayer.cpp

```diff
@@ -82,30 +82,24 @@ void ToolPageLayer::addSpecialButtonTop() {
     createSeparator();
 }
 
-void ToolPageLayer::selectLayer(int layerId) { lc->switchToLay(layerId); }
+void ToolPageLayer::selectLayer(size_t layerId) { lc->switchToLay(layerId); }
 
 void ToolPageLayer::layerMenuClicked(GtkWidget* menu) {
     if (inMenuUpdate) {
         return;
     }
 
-    int layerId = -1;
+    auto it = std::find_if(layerItems.begin(), layerItems.end(),
+                           [menu](std::pair<const size_t, GtkWidget*>& a) { return a.second == menu; });
 
-    for (auto& kv: layerItems) {
-        if (kv.second == menu) {
-            layerId = kv.first;
-            break;
-        }
-    }
-
-    if (layerId < 0) {
+    if (it == layerItems.end()) {
         g_warning("Invalid Layer Menu selected - not handled");
         return;
     }
-
+    auto layerId = it->first;
```

Comment:
Replace with std::find. Please double check what I wrote doesn't change the behaviour.